chat-server/server.js

```diff
@@ -3,28 +3,92 @@ import express from "express";
 import cors from "cors";
 import path from "path";
 import { fileURLToPath } from "url";
+// Get __dirname in ES module
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
 const app = express();
 
 app.use(cors());
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json())
+app.use(express.static(__dirname));
+
 
-// Get __dirname in ES module
-const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
+const newMessages = [
+  {
+    id: 1,
+    from: "Peter",
+    text: "Hello, CYFers!",
+  },
+  {
+    id: 2,
+    from: "Gary",
+    text: "How are you?",
+  }
+]
+
 //This array is our "data store".
 //We will start with one message in the array.
-const messages = [welcomeMessage];
+const messages = [welcomeMessage, ...newMessages];
+
+// GET routes
 
 app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+  response.sendFile(`${__dirname}/index.html`);
+});
+
+app.get("/messages", (request, response) => {
+  response.json(messages)
+});
+
+app.get("/messages/:messageId", (request, response) => {
+  const messageId = parseInt(request.params.messageId);
+  const message = messages.find((message) => message.id === messageId);
+  if (message) {
+    response.json(message);
+  } else {
+    response.status(404).json("Message not found");
+  }
+});
+
+// POST routes
+
+app.post("/messages", (request, response) => {
+  const newMessage = request.body;
+  const { from, text } = newMessage;
+  if (!from || !text) {
+    response.status(400).json("Message must have a Name and a Text");
+    return;
+  }
+
+  const lastId = messages[messages.length - 1].id;
+  newMessage.id = lastId + 1;
+  messages.push(newMessage);
+
+  response.sendFile(`${__dirname}/index.html`);
```

Comment:
This is an interesting response to send in response to POSTing a new message - can you talk through why this is the thing you're sending back? What are you expecting the client to do with this information?

Reply:
Sending the index.html to the client allows the user to see the page with their message already added. I believe this is more useful than simply displaying a message like "Message added" or similar.